### PR TITLE
Add new 'default_group' param for remote_user auth

### PR DESCRIPTION
**Pull request by abretaud**
Hi,
As discussed in https://github.com/galaxy-genome-annotation/dockerized-gmod-deployment/issues/6, a patch to add a new param for RemoteUserAuthenticatorService to allow adding automatically new users to a group.
I don't know if it's the best way to implement it, but it works :)
Anthony

### CODE REVIEW

**Comment by abretaud** [comment]
Oh, and I configure it like this:

```
    authentications = [
        [
            "name":"Remote User Authenticator",
            "className":"remoteUserAuthenticatorService",
            "active":true,
            "params":["default_group": "annotators"]
        ],
        [
            "name":"Username Password Authenticator",
            "className":"usernamePasswordAuthenticatorService",
            "active":true,
            "params":[]
        ]
    ]
```

**Comment by nathandunn** [comment]
Nice.  It looks good.  I'll try to get it tested soon.

File: grails-app/services/org/bbop/apollo/authenticator/RemoteUserAuthenticatorService.groovy [hunk 1]
```diff
@@ -61,6 +64,22 @@ class RemoteUserAuthenticatorService implements AuthenticatorService {
                 log.debug "adding role: ${role}"
                 user.addToRoles(role)
                 role.addToUsers(user)
+
+                if (this.default_group != null) {
```
Comment (nathandunn) [inline grails-app/services/org/bbop/apollo/authenticator/RemoteUserAuthenticatorService.groovy]: In groovy you can do:

if(this.default_group)

File: grails-app/services/org/bbop/apollo/authenticator/RemoteUserAuthenticatorService.groovy [hunk 2]
```diff
@@ -61,6 +64,22 @@ class RemoteUserAuthenticatorService implements AuthenticatorService {
...
+                    log.debug "adding user to default group: ${this.default_group}"
+                    UserGroup userGroup = UserGroup.findByName(this.default_group)
+
+                    if (userGroup == null) {
```
Comment (nathandunn) [inline grails-app/services/org/bbop/apollo/authenticator/RemoteUserAuthenticatorService.groovy]: I like to use the elvis syntax here, but this is fine as well. 

userGroup = userGroup ?: new UserGroup (...).save(...)

**Comment by nathandunn** [comment]
I think it looks good overall.  Just had some fairly minor comments.   Sorry for the delay on this.  Usually I bring in small changes more quickly.

**Comment by nathandunn** [comment]
I can merge these and make the changes, but it would probably be better for your workflow if you do it from your end.   Let me know what makes sense to you. 

Cheers and thanks!

**Comment by abretaud** [comment]
Thanks, changes are done!

**Comment by nathandunn** [comment]
@abretaud  I added a small set of stuff just to keep internal standards set.

**Comment by nathandunn** [comment]
But should be functionally identical.

**Comment by abretaud** [comment]
No problem, thanks!

**Pull request merged**